[quant][doc] Adding a table comparing eager and fx graph mode

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50413 [quant][doc] Adding a table comparing eager and fx graph mode**

Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

jerryzh168 committed:
[quant][doc] Adding a table comparing eager and fx graph mode

Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

raghuramank100 commented on docs/source/quantization.rst:
```diff
@@ -84,7 +84,42 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
+
+The following table compares the differences between Eager Mode Quantization and FX Graph Mode Quantization:
+
++-----------------+-------------------+-------------------+
```
Do we also want to call out that both methods support static/dynamic and QAT? Is Fx support for dynamic quant on par with that of eager?

jerryzh168 replied:
Sure, yeah FX support is on par with eager

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
+|Quantizing       |Supported          |Supported          |
+|Modules          |                   |                   |
++-----------------+-------------------+-------------------+
+|Quantizing       |Need Refactor      |Automatic          |
```
optional: `Manual` instead of `Need Refactor`, to match the others?

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
+|                 |Training:          |Training:          |
+|                 |Static             |Static             |
++-----------------+-------------------+-------------------+
+|Expect to work on|Yes                |No (We'll integrate|
+|arbitrary model  |                   |FX Graph Mode      |
+|                 |                   |Quantization into  |
+|                 |                   |domain libraries   |
+|                 |                   |like torchvision)  |
++-----------------+-------------------+-------------------+
```
I could see people having questions about what `works on an arbitrary model` means, it's possible to interpret this as "quantization will always work on any model without an accuracy drop".  Should we either remove this line or provide some clarifying details?

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
+|                 |                   |domain libraries   |
+|                 |                   |like torchvision)  |
++-----------------+-------------------+-------------------+
+|Input/Output     |``torch.nn.Module``|``torch.nn.Module``|
+|Model Type       |                   |(May need some     |
+|                 |                   |refactors to make  |
+|                 |                   |the model          |
+|                 |                   |compatible with FX |
+|                 |                   |Graph Mode         |
+|                 |                   |Quantization)      |
++-----------------+-------------------+-------------------+
```
optional: `symbolically traceable torch.nn.Module`?

jerryzh168 replied:
I want to avoid mentioning symbolic traceability actually

vkuzo reviewed (approved):
awesome!

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][doc] Adding a table comparing eager and fx graph mode"


Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

[ghstack-poisoned]
